File: java/altgraph_web_app/src/main/java/org/cjoakim/cosmos/altgraph/data/common/repository/LibraryRepository.java
package org.cjoakim.cosmos.altgraph.data.common.repository;

import com.azure.spring.data.cosmos.repository.CosmosRepository;
import org.cjoakim.cosmos.altgraph.data.common.model.npm.Library;
import org.springframework.stereotype.Component;
import org.springframework.stereotype.Repository;

/**
 * This class is a Spring Data Repository for Cosmos DB NPM Library documents.
 * Chris Joakim, Microsoft, November 2022
 */

@Component
@Repository
public interface LibraryRepository extends CosmosRepository<Library, String> {

    Iterable<Library> findByPkAndTenant(String pk, String tenant);

    Iterable<Library> findByPkAndTenantAndDoctype(String pk, String tenant, String doctype);

}

File: dotnet/src/altgraph_shared_app/Repositories/Npm/LibraryRepository.cs
using altgraph_shared_app.Models;
using altgraph_shared_app.Models.Npm;
using Microsoft.Azure.CosmosRepository;

namespace altgraph_shared_app.Services.Repositories.Npm
{
  public class LibraryRepository
  {
    public IRepository<Library> Libraries { get; private set; }
    public LibraryRepository(IRepository<Library> libraries)
    {
      Libraries = libraries;
    }
    public async Task<IEnumerable<Library>> FindByPkAndTenantAsync(string pk, string tenant)
    {
      return await Libraries.GetAsync(x => x.Pk == pk && x.Tenant == tenant);
    }

    public async Task<IEnumerable<Library>> FindByPkAndTenantAndDoctypeAsync(string pk, string tenant, string doctype)
    {
      return await Libraries.GetAsync(x => x.Pk == pk && x.Tenant == tenant && x.Doctype == doctype);
    }
  }
}